frontend/src/components/Header/StopClock.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import React, { Component } from 'react';
+import CircularProgressbar from 'react-circular-progressbar';
+import 'react-circular-progressbar/dist/styles.css';
+
+type Props = {
+  startTime: number, // Date.now() ms at start
+  countdown: number, // delta in seconds
+  currentTime: number, // Date.now() at present
+  history: *,
+};
+
+type State = {
+  currentTime: number,
+};
+
+type CircularProgressbarProps = {
+  text: string,
+  percentage: number,
+};
+
+const StyledCircularProgressbar = (props: CircularProgressbarProps) => (
+  <CircularProgressbar
+    percentage={props.percentage}
+    text={props.text}
+    strokeWidth={10}
+    styles={{
+      root: {},
+      path: {
+        stroke: '#1c8bf4',
+        strokeLinecap: 'butt',
+        transition: 'stroke-dashoffset 0.5s ease 0s',
+      },
+      trail: {
+        stroke: '#e6e6e6',
+      },
+      text: {
+        fill: '#1c8bf4',
+        fontSize: '30px',
+      },
+    }}
+  />
+);
+
+const getPercRemaining = (
+  currentTime: number,
+  startTime: number,
+  countdown: number
+) => {
+  return (currentTime - startTime) / (countdown * 1000);
+};
+
+const getTimeLabel = (secondsRemaining: number) =>
+  secondsRemaining < 60
+    ? `${Math.ceil(secondsRemaining)}s`
+    : `${Math.ceil(secondsRemaining / 60)}m`;
+
+class StopClock extends Component<Props, State> {
+  state: State = {
+    currentTime: Date.now(),
+  };
+
+  tick = () => {
+    this.setState({
+      currentTime: Date.now(),
+    });
+  };
+
+  timer = null;
+  endTime = 0;
+  componentDidMount() {
+    this.timer = setInterval(this.tick, 500);
+    this.endTime = this.props.startTime + 1000 * this.props.countdown;
```

Comment:
There is already an setInterval setup for the timer in https://github.com/cybsafe/phishtray/blob/master/frontend/src/App.js#L96-L100

Also the exercise time should be kept in redux because multiple components in the app might care about it.

Reply:
It doesn't matter if the `tickTimer` method is fired a lot because only the components that care about it changing will actually update. The clock would be one of those components but there could be any number of other components in the app that would care about the current time.